samtranslator/intrinsics/actions.py

```diff
@@ -503,3 +504,46 @@ def _get_resolved_dictionary(self, input_dict, key, resolved_value, remaining):
             input_dict[key] = [resolved_value] + remaining
 
         return input_dict
+
+
+class FindInMapAction(Action):
+    """
+    This action can't be used along with other actions.
+    """
+    intrinsic_name = "Fn::FindInMap"
+
+    def resolve_parameter_refs(self, input_dict, parameters):
+        """
+        Recursively resolves "Fn::FindInMap"references that are present in the mappings and returns the value.
+        If it is not in mappings, this method simply returns the input unchanged.
+
+        :param input_dict: Dictionary representing the FindInMap function. Must contain only one key and it
+                           should be "Fn::FindInMap".
+
+        :param parameters: Dictionary of mappings from the SAM template
+        """
+        if not self.can_handle(input_dict):
+            return input_dict
+
+        value = input_dict[self.intrinsic_name]
+
+        # FindInMap expects an array with 3 values
+        if not isinstance(value, list) or len(value) != 3:
```

Comment:
Should we also throw errors in the other scenarios where we `return input_dict`? Or are there cases where we can't statically resolve them but they may get successfully resolved at runtime during CFN change set execution?

Reply:
You are right. In other scenarios, it is possible that some intrinsic functions are still left unresolved.